hathor/utils/list.py

```diff
@@ -0,0 +1,24 @@
+#  Copyright 2023 Hathor Labs
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from typing import List, Optional, TypeVar
+
+T = TypeVar('T')
+
+
+def single_or_none(_list: List[T]) -> Optional[T]:
+    """Function to convert a list with at most one element to the given element or None."""
```

Comment:
This type of simple function is great for using in-doc tests, like:

```python
def single_or_none(_list: List[T]) -> Optional[T]:
    """ ...
    >>> single_or_none([1])
    1
    >>> single_or_none([]) is None
    True
    >>> single_or_none([1, 2])
    Traceback (most recent call last):
     ...
    AssertionError: expected one value at most
    """
    # ...
```

Or something like that, you can grep for `>>>` to see some other places we do this. The great thing is that it works not only as documentation but also as unittests.

Reply:
Nice, I didn't know about this. Done!